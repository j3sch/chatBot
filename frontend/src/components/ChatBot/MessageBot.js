function Index({ text, img }) {
	return (
		<div data-testid="id2" className="flex flex-row mt-4 items-center ml-4 chatBotMessage">
			<img src={img} alt="Business Mann 42" className="h-8 w-8 rounded-full" />
			<div className="pl-4 pr-4 pt-2 pb-2 ml-4 mr-4 bg-white dark:bg-slate-600 rounded-lg drop-shadow-sm">
				<p className="text text-black dark:text-white">{text}</p>
			</div>
		</div>
	);
}

export default Index;
