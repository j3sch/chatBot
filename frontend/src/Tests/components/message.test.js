import { render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import MessageUser from '../../components/ChatBot/MessageUser';
import MessageBot from '../../components/ChatBot/MessageBot';

afterEach(() => {
	cleanup();
});

test('MessageUser test', () => {
	render(<MessageUser text="das ist die Nachricht von einem User" img="test-image.jpg" />);
	const element = screen.getByTestId('id1');
	expect(element).toBeInTheDocument();
	expect(element).toHaveTextContent('das ist die Nachricht von einem User');
});

test('MessageBot test', () => {
	render(<MessageBot text="das ist die Nachricht von einem Bot" img="test-image.jpg" />);
	const element = screen.getByTestId('id2');
	expect(element).toBeInTheDocument();
	expect(element).toHaveTextContent('das ist die Nachricht von einem Bot');
});

test('matches snapshot MessageUser', () => {
	const tree = renderer
		.create(<MessageUser text="das ist die Nachricht von einem User" img="test-image.jpg" />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

test('matches snapshot MessageBot', () => {
	const tree = renderer
		.create(<MessageBot text="das ist die Nachricht von einem Bot" img="test-image.jpg" />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
