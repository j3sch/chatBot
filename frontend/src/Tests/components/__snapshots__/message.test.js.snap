// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot MessageBot 1`] = `
<div
  className="flex flex-row mt-4 items-center ml-4 chatBotMessage"
  data-testid="id2"
>
  <img
    alt="Business Mann 42"
    className="h-8 w-8 rounded-full"
    src="test-image.jpg"
  />
  <div
    className="pl-4 pr-4 pt-2 pb-2 ml-4 mr-4 bg-white dark:bg-slate-600 rounded-lg drop-shadow-sm"
  >
    <p
      className="text text-black dark:text-white"
    >
      das ist die Nachricht von einem Bot
    </p>
  </div>
</div>
`;

exports[`matches snapshot MessageUser 1`] = `
<div
  className="flex flex-row mt-4 items-center ml-4 justify-end userMessage"
  data-testid="id1"
>
  <div
    className="pl-4 pr-4 pt-2 pb-2 ml-4 mr-4 bg-indigo-600 rounded-lg drop-shadow-sm"
  >
    <p
      className="text text-white"
    >
      das ist die Nachricht von einem User
    </p>
  </div>
</div>
`;
